File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_56_correct_bracketing.lua
-- brackets is a string of "<" and ">".
-- return True if every opening bracket has a corresponding closing bracket.
-- >>> correct_bracketing("<")
-- False
-- >>> correct_bracketing("<>")
-- True
-- >>> correct_bracketing("<<><>>")
-- True
-- >>> correct_bracketing("><<>")
-- False
local function correct_bracketing(brackets)
  local opening_brackets = brackets:gsub("[<>]", "")
  local closing_brackets = brackets:gsub("[<>]", "")
  local opening_brackets_count = opening_brackets:len()
  if opening_brackets_count ~= closing_brackets:len() then
    return false
  end
  for i = 1, opening_brackets_count do
    local opening_bracket = opening_brackets:sub(i, i)
    local closing_bracket = closing_brackets:sub(i, i)
    if opening_bracket ~= closing_bracket then
      return false
    end
  end
  return true
end


lu = require('luaunit')

function test_humaneval()
local candidate = correct_bracketing
    lu.assertEquals(candidate('<>'), true)
    lu.assertEquals(candidate('<<><>>'), true)
    lu.assertEquals(candidate('<><><<><>><>'), true)
    lu.assertEquals(candidate('<><><<<><><>><>><<><><<>>>'), true)
    lu.assertEquals(candidate('<<<><>>>>'), false)
    lu.assertEquals(candidate('><<>'), false)
    lu.assertEquals(candidate('<'), false)
    lu.assertEquals(candidate('<<<<'), false)
    lu.assertEquals(candidate('>'), false)
    lu.assertEquals(candidate('<<>'), false)
    lu.assertEquals(candidate('<><><<><>><>><<>'), false)
    lu.assertEquals(candidate('<><><<><>><>>><>'), false)
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_56_correct_bracketing.R
# brackets is a string of "<" and ">".
# return True if every opening bracket has a corresponding closing bracket.
# >>> correct_bracketing("<")
# False
# >>> correct_bracketing("<>")
# True
# >>> correct_bracketing("<<><>>")
# True
# >>> correct_bracketing("><<>")
# False
correct_bracketing <- function(brackets) {
    stack <- list()
    for (i in 1:length(brackets)) {
        if (brackets[i] == "<") {
            stack <- c(stack, "<")
        } else {
            if (length(stack) == 0) {
                return(FALSE)
            }
            stack <- stack[-1]
        }
    }
    if (length(stack) == 0) {
        return(TRUE)
    } else {
        return(FALSE)
    }
}
correct_bracketing("<")
correct_bracketing("<>")
correct_bracketing("<<><>>")
correct_bracketing("><<>")


test_humaneval <- function() {
candidate <- correct_bracketing
    if(!identical(candidate('<>'), TRUE)){quit('no', 1)}
    if(!identical(candidate('<<><>>'), TRUE)){quit('no', 1)}
    if(!identical(candidate('<><><<><>><>'), TRUE)){quit('no', 1)}
    if(!identical(candidate('<><><<<><><>><>><<><><<>>>'), TRUE)){quit('no', 1)}
    if(!identical(candidate('<<<><>>>>'), FALSE)){quit('no', 1)}
    if(!identical(candidate('><<>'), FALSE)){quit('no', 1)}
    if(!identical(candidate('<'), FALSE)){quit('no', 1)}
    if(!identical(candidate('<<<<'), FALSE)){quit('no', 1)}
    if(!identical(candidate('>'), FALSE)){quit('no', 1)}
    if(!identical(candidate('<<>'), FALSE)){quit('no', 1)}
    if(!identical(candidate('<><><<><>><>><<>'), FALSE)){quit('no', 1)}
    if(!identical(candidate('<><><<><>><>>><>'), FALSE)){quit('no', 1)}
}
test_humaneval()